### DISCUSSION TITLE
Incredible RFC

### DISCUSSION THREAD

**Discussion by kirawi** [helix-editor/helix#10096] [Ideas]
To celebrate the end of the first quarter of 2024, I've elected myself to spend day and night responding to some of the dismay regarding the plugin system in order to create something I think everyone will be pleased with.

This new plugin system, called "Reactive User Script Thing" or RUST for short (not to be confused with the game Rust), would use the latest in LLM research to take call the appropriate API functions based on the input. This allows any language to be used, including but not limited to, Elvish and doctor's handwriting.

I urge everyone to consider the benefits and implications of such a shift compared to our predecessors!

(check what day it is)

**Comment by pascalkuthe** [comment]
This can be combined with the new LLM integration of rust-lang  so we can stay *current*

https://github.com/rust-lang/rfcs/pull/3603

**Reply by kirawi** [reply]
That's amazing! I wonder if they would also be open to an RFC to integrate blockchain technology™ to std. I was thinking of having the plugin marketplace be built on top of it for maximum user security.
